dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go

```diff
@@ -26,16 +26,49 @@ import (
 	"github.com/GoogleCloudPlatform/k8s-config-connector/dev/tools/controllerbuilder/pkg/protoapi"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/dev/tools/controllerbuilder/scaffold"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/klog/v2"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 )
 
 type GenerateCRDOptions struct {
 	*options.GenerateOptions
 
 	OutputAPIDirectory string
-	ResourceKindName   string
-	ResourceProtoName  string
+
+	Resources ResourceList
+}
+
+type Resource struct {
+	Kind      string
+	ProtoName string
+}
+
+type ResourceList []Resource
+
+var _ pflag.Value = &ResourceList{}
+
+func (r *ResourceList) Type() string { return "resources" }
+
+func (r *ResourceList) String() string {
+	var sb strings.Builder
+	for _, res := range *r {
+		fmt.Fprintf(&sb, "%s:%s", res.Kind, res.ProtoName)
+	}
+	return sb.String()
+}
+
+func (r *ResourceList) Set(s string) error {
+	tokens := strings.Split(s, ":")
+	if len(tokens) != 2 || tokens[0] == "" || tokens[1] == "" {
+		return fmt.Errorf("expected [KRMKind]:[ProtoResourceName], got %q", s)
+	}
+	*r = append(*r, Resource{
+		Kind:      tokens[0],
+		ProtoName: tokens[1],
```

Comment:
nit: previously, users weren’t required to input a capitalized proto resource name, as the tool automatically called `capitalizeFirstRune` to handle this for them.

Reply:
Yeah, I think that was a bit too user-friendly :-)